src/pystatis/table.py

```diff
@@ -133,6 +134,14 @@ def parse_genesis_table(data: pd.DataFrame) -> pd.DataFrame:
         # Extracts time column with name from first element of Zeit_Label column
         time = pd.DataFrame({data["Zeit_Label"].iloc[0]: data["Zeit"]})
 
+        # Some tables of Genesis can have a regional code (AGS) as first attribute
+        ags_code = None
+        pos_of_ags_col = np.where(data.iloc[0].isin(config.REGIO_AGS_CODES))[0]
+        if pos_of_ags_col.size > 0:
+            pos_of_ags_col = pos_of_ags_col[0]
+            label = "Amtlicher Gemeindeschlüssel (AGS)"  # en: official municipality code (AGS)
+            ags_code = pd.Series(data=data.iloc[:, pos_of_ags_col + 2], name=label)
+
```

Comment:
A general comment regarding all three parsing functions: With currently increasing parsing functionality (in other issues as well), a lot of code is repeated amongst the three parsing functions, and especially the regio and genesis one. Should we try to abstract common behaviour away in to smaller, special purpose subfunctions?

Reply:
I don't see a benefit from this. It is true that it is harder to maintain but abstracting it away in a lot of small helper functions or classes only increases abstraction and makes it harder to come back to. We have three dedicated methods and if you need to change something you know exactly where to look for. Finding a good abstraction is really hard and there will ALWAYS be small differences in the three databases for example like the column names so you cannot easily abstract this away, we would need something like three Prettify classes that can be entered here and a factory method or something to instantiate the right processing class. A lot of boilerplate code to handle this simple logic

Reply:
But we should definitely refactor once we have a good idea and way forward and a more stable code. Right now I would definitely not start with this until we have Zensus2022 live